### PR TITLE
Ensure that text labels in nested controls are translated correctly

### PR DESCRIPTION
**Pull request by lloyddewit**
Partially fixes issue #3721
Partially fixes issue #6454 

Issue #3721 states:
'Sometimes we have controls with the same name in the same form/dialog/subdialog. In most cases, this can be resolved by changing the name of the controls to make them unique. However, there are some cases where this is not possible. An example is in sdgLayerOptions. This subdialog contains 2 ucrGeom objects, which both have a lblTypeofLayer control. Only one of these is translated in the current translation mechanism. In order to translate both labels, we need a way to uniquely identify and store each label. Another sympton of this issue is that checkboxes do not translate correctly when the user toggles the language.'

This PR aims aims to fix this issue.

@rdstern Please could you test? thanks

### CODE REVIEW

**Comment by lloyddewit** [comment]
@rdstern This PR now includes the latest French translation updates from @N-thony PR #6468.

**Comment by rdstern** [comment]
I am starting to look again systematically down the dialogues - starting with Prepare.
a) Prepare > Data Frame > View Data:  The radio buttons in French don't have the complete labels.  Of course they could be made wider, but there seems to be enough space within the existing buttons to add the word donnees.  So I wonder if there is a (general?) option in the code/button/function to permit the text in such buttons to have less space at the start/end?
image

b) Next dialogue Prepare > Data Frame > Rename Column is perhaps the first example where the current translation is (almost) perfect.  Here it is:
image

So I am becoming really "picky" now.  In the main (bottom) control, could the Comment field be just slightly smaller, so it starts just a tad to the right, and the colon after Commentaire: is then fully visible.  (I wouldn't be so concerned usually, but it is a single change (I think?) which will then be implemented automatically on every dialogue.

**Comment by lloyddewit** [comment]
@rdstern Thank you for the comments. I expect that the longer French text will cause presentation and position problems in other dialogs also.

I suggest that:
- you verify that the translation **mechanism** is working as expected (i.e. it is translating text into French, even if the French causes presentation/position problems). If you can verify this then please could you approve this PR?
- @shadrackkibet peer reviews and merges this PR according to our normal process. I'd like to get this PR merged as soon as possible so that I can add the next translation mechanism improvements (tool tips, dropdown lists and control tags for R parameters). 
- one of the developers fixes the presentation/position problems you already identified here, and in issue #6454, issue #6464 and issue #6465. These fixes can be done in a new PR.
- one (or more) of the developers systematically goes through the dialogs and fixes the presentation/position problems. these fixes can be done in one or more new PRs.

**Comment by rdstern** [comment]
@lloyddewit I have now just seen your comment.

So I now looked quickly at the x and y axis thingy and that seems fine.  So, do you want to examine the single tricky item above now, or shall I approve and you also look at that "feature" in the next pull request?

**Comment by N-thony** [comment]
> @rdstern Thank you for the comments. I expect that the longer French text will cause presentation and position problems in other dialogs also.
> 
> I suggest that:
> 
> * you verify that the translation **mechanism** is working as expected (i.e. it is translating text into French, even if the French causes presentation/position problems). If you can verify this then please could you approve this PR?
> * @shadrackkibet peer reviews and merges this PR according to our normal process. I'd like to get this PR merged as soon as possible so that I can add the next translation mechanism improvements (tool tips, dropdown lists and control tags for R parameters).
> * one of the developers fixes the presentation/position problems you already identified here, and in issue #6454, issue #6464 and issue #6465. These fixes can be done in a new PR.
> * one (or more) of the developers systematically goes through the dialogs and fixes the presentation/position problems. these fixes can be done in one or more new PRs.

@lloyddewit I suggest after this being merged, to work on the Climatic menu in one PR.

**Review by rdstern — APPROVED** [review]
@lloyddewit I am approving, because the only minor quirk I found (and reported in the pull request) could also be fixed in the next round

File: instat/sdgOpenNetCDF.vb [hunk 1]
```diff
@@ -24,14 +24,19 @@ Public Class sdgOpenNetCDF
...
+    ' If we don't do this then, when we build the translations database, a null reference exception 
+    ' is triggered in `sdgOpenNetCDF_Closing()`.
+    Private lstMinTextBoxes As List(Of ucrInputTextBox) = New List(Of ucrInputTextBox)
+    Private lstMaxTextBoxes As List(Of ucrInputTextBox) = New List(Of ucrInputTextBox)
+    Private lstAxesLabels As List(Of Label) = New List(Of Label)
+    Private lstMinLabels As List(Of Label) = New List(Of Label)
+    Private lstMaxLabels As List(Of Label) = New List(Of Label)
+    Private lstDims As List(Of String) = New List(Of String)
+    Private lstAxesDetected As List(Of Boolean) = New List(Of Boolean)
+    Private lstFunctions As List(Of RFunction) = New List(Of RFunction)
```
Comment (shadrackkibet) [inline instat/sdgOpenNetCDF.vb:38]: I am happy with these initialisations. Simply use this format for consistency? 
`Private lstMinTextBoxes As New List(Of ucrInputTextBox)`

**Review by shadrackkibet — APPROVED** [review]
@lloyddewit only one minor comment. If you wish to change let me know, otherwise I am happy to merge this.

File: instat/sdgOpenNetCDF.vb (hunk 1 above)
Comment (lloyddewit) [reply instat/sdgOpenNetCDF.vb]: Yes, I agree. I changed the code

**Comment by lloyddewit** [comment]
@rdstern You passed the 'deliberate test' ;) - this is a real bug. 
I investigated. It happens because the dialog is implemented in a non-standard way. It translates as the first step in the load process. This is before some of the control text is set, so for some controls, there is nothing yet to translate. When the dialog is opened the second time, the control text has already been initialized so it can be translated. To solve the issue is a one-line change: move the translation line to the end of the load function. I did this in the Sort dialog to demonstrate. The Sort dialog should now initialize correctly.
As you discovered, there are other dialogs with a similar problem. I would like to solve these together in a new PR.
If you agree, please could you test the Sort dialog and approve if you're happy?

@shadrackkibet Thank you for the peer review and comment. I resolved. If @rdstern approves, then please could you also approve/merge? 

Thanks!

**Comment by lloyddewit** [comment]
> 
> 
> > @rdstern Thank you for the comments. I expect that the longer French text will cause presentation and position problems in other dialogs also.
> > I suggest that:
> > 
> > * you verify that the translation **mechanism** is working as expected (i.e. it is translating text into French, even if the French causes presentation/position problems). If you can verify this then please could you approve this PR?
> > * @shadrackkibet peer reviews and merges this PR according to our normal process. I'd like to get this PR merged as soon as possible so that I can add the next translation mechanism improvements (tool tips, dropdown lists and control tags for R parameters).
> > * one of the developers fixes the presentation/position problems you already identified here, and in issue #6454, issue #6464 and issue #6465. These fixes can be done in a new PR.
> > * one (or more) of the developers systematically goes through the dialogs and fixes the presentation/position problems. these fixes can be done in one or more new PRs.
> 
> @lloyddewit I suggest after this being merged, to work on the Climatic menu in one PR.

@N-thony Thanks, good idea
Please could you try and resolve the remaining parts of issue #6454, issue #6464 and issue #6465; and then start checking/updating the dialogs in the Climatic menu?

@rdstern Is that OK from your side or did you have a different task you'd prefer @N-thony to work on?

**Review by shadrackkibet — APPROVED** [review]
@rdstern if you approve then we can merge.

**Review by rdstern — APPROVED** [review]
@lloyddewit  the sort now works as you said.  And I was pleased that the row numbers doesn't yet work.
I see that @shadrackkibet has approved.  I am happy for it to be merged, before the other dialogue us corrected.  
I will follow your message and write separately about the general issue.

**Pull request merged**